src/transaction-flow/transaction/migrateProfileWithSync.ts

```diff
@@ -0,0 +1,101 @@
+import { Profile, PublicENS, RecordItem, TransactionDisplayItem } from '@app/types'
+import { recordItemToKeyValue } from '@app/utils/editor'
+import { RecordOptions } from '@ensdomains/ensjs/dist/cjs/utils/recordHelpers'
+import type { JsonRpcSigner } from '@ethersproject/providers'
+import { contentHashToString } from '../../utils/contenthash'
+
+type Data = {
+  name: string
+  records?: RecordOptions
+}
+
+const displayItems = ({ name }: Data): TransactionDisplayItem[] => [
+  {
+    label: 'name',
+    value: name,
+    type: 'name',
+  },
+]
+
+export const syncRecords = (
+  before?: RecordItem[],
+  after?: RecordItem[],
+  overwrites?: { key: string; value: string }[],
+) => {
+  const beforeArr = before ? before.map(recordItemToKeyValue) : []
+  const beforeHash =
+    beforeArr.reduce<{ [key: string]: string }>((hash, item) => {
+      const key = item.key as string
+      const value = item.value as string
+      const newHash = { ...hash, [key]: value }
+      return newHash
+    }, {}) || {}
+
+  return Object.values(
+    [
+      ...beforeArr.map(({ key }) => ({ key, value: '' })),
+      ...(after ? after.map(recordItemToKeyValue) : []),
+      ...(overwrites || []),
+    ].reduce<{
+      [key: string]: { key: string; value: string }
+    }>((acc, text) => {
+      const key = text.key as string
+      const value = text.value as string
+      if (beforeHash[key] === value) {
+        delete acc[key]
+      } else {
+        acc[key] = { key, value }
+      }
+      return acc
+    }, {}),
+  )
+}
+
+const transaction = async (signer: JsonRpcSigner, ens: PublicENS, data: Data) => {
+  const profile = await ens.getProfile(data.name)
+  if (!profile) throw new Error('No profile found')
+  if (!profile.records) throw new Error('No records found')
+  const resolverAddress = (await ens.contracts!.getPublicResolver()!).address
+
+  let resolverProfile: Profile | undefined
+  if (profile.resolverAddress !== resolverAddress) {
+    resolverProfile = await ens.getProfile(data.name, {
+      resolverAddress,
+    })
+  }
+
+  const contentHash = (() => {
+    if (data.records?.contentHash || data.records?.contentHash === '')
+      return contentHashToString(data.records.contentHash)
+    const afterCh = contentHashToString(profile.records.contentHash)
+    const beforeCh = contentHashToString(resolverProfile?.records?.contentHash)
+    if (beforeCh && !afterCh) return ''
+    if (afterCh) return afterCh
+    return undefined
+  })()
```

Comment:
any reason this is a func and being called instead of just being inline?

Reply:
Over thought logic...